nodedb.go

```diff
@@ -584,119 +508,18 @@ func (ndb *nodeDB) DeleteFastNode(key []byte) error {
 
 // deleteNodesFrom deletes the given node and any descendants that have versions after the given
 // (inclusive). It is mainly used via LoadVersionForOverwriting, to delete the current version.
-func (ndb *nodeDB) deleteNodesFrom(version int64, hash []byte) error {
-	if len(hash) == 0 {
-		return nil
-	}
-
-	node, err := ndb.unsafeGetNode(hash)
-	if err != nil {
-		return err
-	}
-
-	if node.version < version {
-		// We can skip the whole sub-tree since children.version <= parent.version.
-		return nil
-	}
-
-	if node.leftHash != nil {
-		if err := ndb.deleteNodesFrom(version, node.leftHash); err != nil {
-			return err
-		}
-	}
-	if node.rightHash != nil {
-		if err := ndb.deleteNodesFrom(version, node.rightHash); err != nil {
-			return err
-		}
-	}
-
-	if node.version >= version {
-		if err := ndb.batch.Delete(ndb.nodeKey(hash)); err != nil {
-			return err
-		}
-
-		ndb.nodeCache.Remove(hash)
-	}
-
-	return nil
-}
-
-// Saves orphaned nodes to disk under a special prefix.
-// version: the new version being saved.
-// orphans: the orphan nodes created since version-1
-func (ndb *nodeDB) SaveOrphans(version int64, orphans map[string]int64) error {
-	ndb.mtx.Lock()
-	defer ndb.mtx.Unlock()
-
-	toVersion, err := ndb.getPreviousVersion(version)
-	if err != nil {
-		return err
-	}
-
-	for hash, fromVersion := range orphans {
-		logger.Debug("SAVEORPHAN %v-%v %X\n", fromVersion, toVersion, hash)
-		err := ndb.saveOrphan([]byte(hash), fromVersion, toVersion)
-		if err != nil {
-			return err
+func (ndb *nodeDB) deleteNodesFrom(version int64, root []byte) error {
+	return ndb.traverseTree(root, func(node *Node) (bool, error) {
+		if node.version < version {
```

Comment:
```suggestion
        // We can skip the whole sub-tree since children.version <= parent.version.
		if node.version < version {
```

Propose to maintain this comment since it's pretty helpful when reading.